controlnext/views.py

```diff
@@ -67,17 +55,29 @@ def post(self, request, random_url_slug):
         return RestResponse(request.POST)
 
 
+class Http403View(TemplateView):
+    template_name = '403.html'
+    page_title = '403 Error'
+
+
 class BasinView(UiView):
     template_name = 'controlnext/basin_detail.html'
     page_title = _('ControlNEXT')
 
     def get(self, request, random_url_slug, *args, **kwargs):
-        self.basin = find_basin(random_url_slug)
-        self.random_url_slug = random_url_slug
-        if (self.basin.grower.crop and
-                is_valid_crop_type(self.basin.grower.crop)):
-            self.crop_type = self.basin.grower.crop
-        return super(BasinView, self).get(request, *args, **kwargs)
+        basins_query = GrowerInfo.objects.filter(
+            userprofile__user=request.user)
+        basins = Basin.objects.filter(grower__in=basins_query)
+        slugs = [str(b.random_url_slug) for b in basins]
+        if random_url_slug in slugs:
+            self.basin = find_basin(random_url_slug)
+            self.random_url_slug = random_url_slug
+            if (self.basin.grower.crop and
+                    is_valid_crop_type(self.basin.grower.crop)):
+                self.crop_type = self.basin.grower.crop
+            return super(BasinView, self).get(request, *args, **kwargs)
```

Comment:
Oei, je roept hier pas super aan? Dan lijkt `self.basin` inderdaad een niet geheel zij-effect-vrij iets te zijn.

Als je eens iets doet zoals dit:

    @property
    def basin(self):
        # De code die de basin ophaalt, makkelijk overschrijfbaar in een subclass.